### ISSUE TITLE
Errors when using roleArn in providerCredentialOpts

### ISSUE THREAD

**Issue by jam13** [pulumi/pulumi-eks#418]
I have a working EKS cluster that uses `providerCredentialOpts.profileName` for cluster auth. I've added roleMappings to a cluster admin IAM role as per https://www.pulumi.com/blog/simplify-kubernetes-rbac-in-amazon-eks-with-open-source-pulumi-packages/ and this works as expected.

When subsequently setting the `providerCredentialOpts.roleArn` to the admin role I get the following error twice during preview and update

```
    An error occurred (AccessDenied) when calling the AssumeRole operation: User: arn:aws:iam::123456789012:user/foo is not authorized to perform: sts:AssumeRole on resource: Calling [toJSON] on an [Output<T>] is not supported.
    To get the value of an Output as a JSON value or JSON string consider either:
        1: o.apply(v => v.toJSON())
        2: o.apply(v => JSON.stringify(v))
    See https://pulumi.io/help/outputs for more details.
    This function may throw in a future version of @pulumi/pulumi.
```

It doesn't appear to break anything, and the stack output of the kubeconfig (`eksCluster.kubeconfig`) has the correct role added to the args, however the errors make me think I'm doing something wrong.

Manually assuming the role in question with the AWS CLI works correctly with no errors.

@pulumi/pulumi 2.8.2
@pulumi/aws 2.13.1
@pulumi/eks: 0.19.3
@pulumi/kubernetes 2.4.2

**Comment by leezen** [comment]
Could you please provide your source code?

**Comment by jam13** [comment]
```
    const assumeRolePolicy = callerIdentity.accountId.apply(accountId => JSON.stringify(
      {
        Version: "2012-10-17",
        Statement: [
          {
            Sid: "",
            Effect: "Allow",
            Principal: {
              AWS: `arn:aws:iam::${accountId}:root`
            },
            Action: "sts:AssumeRole"
          }
        ]
      }
    ));
    const clusterAdminRole = new aws.iam.Role("clusterAdminRole", {
      assumeRolePolicy,
      tags: {
        clusterAccess: "admin-usr"
      }
    });

   new eks.Cluster("cluster", {
      vpcId,
      subnetIds,
      nodeAssociatePublicIpAddress: false,
      skipDefaultNodeGroup: true,
      providerCredentialOpts: {
        profileName: aws.config.profile,
        roleArn: clusterAdminRole.arn
      },
      instanceRole,
      roleMappings: [
        {
          groups: ["system:masters"],
          roleArn: clusterAdminRole.arn,
          username: "pulumi:admin-usr"
        }
      ]
    });
```

**Comment by leezen** [comment]
@metral is this possibly because there's no apply on the `providerCredentialsOpts` in https://github.com/pulumi/pulumi-eks/blob/master/nodejs/eks/cluster.ts#L518?

**Comment by metral** [comment]
Thanks for filing the issue @jam13 

> is this possibly because there's no apply on the providerCredentialsOpts
Yep, this was the source of the bug.

We've got a fix in #421 that was tested using a [similar example setup](https://gist.github.com/metral/94dd554d019505a335de151d1e361eb6) as @jam13 provided.
